### DISCUSSION TITLE
Understanding the idea behind groups and ownership

### DISCUSSION THREAD

**Discussion by sur5r** [inventree/InvenTree#4228] [General]
I'm struggling how the ownership feature is meant to be used with groups.

I have a group for each customer so I can properly assign ownership of parts that belong to customers of mine.
Adding my user as a member to each of these groups gives a warning in the Django admin panel (`The following users are members of multiple groups`) and all stock items end up being read only. So this is clearly not how it's meant to be used.

I'm a bit confused by the fact that a user can only be a member of a single group. What's the idea behind this?

**Comment by miggland** [comment]
To assign stock to customers:

1. add your customers as companies (see the sell menu)
2. assign parts to them. This is achieved either via a Sales Order, or can be done for any stock you register in the system.

**Comment by sur5r** [comment]
Sales orders don't match this scenario. I'm not talking about parts I'm going to sell to customers. This is about parts the customer bought on their behalf which are stored at my place for later assembly.

**Reply by miggland** [reply]
Right, then maybe manually assign them using the function "Assign to Customer"?

It's accessible on a stock item page, eg
https://demo.inventree.org/stock/item/231/
![image](https://user-images.githubusercontent.com/1314853/213232057-1c201fe3-838e-4199-8bcc-1c7751063dba.png)

After assigning, this stock will be shown on the customer's page:
![image](https://user-images.githubusercontent.com/1314853/213232798-96bcf889-4020-4302-b32e-92ade535fc16.png)

**Reply by sur5r** [reply]
Thanks for the hint! I completely overlooked that menu item. It indeed makes more sense for my current use case.

**Reply by miggland** [reply]
Just note that the items won't be shown as "in stock" anymore..

**Reply by sur5r** [reply]
Ah, snap! Back to square one...

**Reply by miggland** [reply]
Yeah sorry, didn't think about that before. The effect is the same as if you sold the items, just without the Sales Order step in between..

Maybe add a location for each customer instead, and put the stock in the corresponding stock location? You can move back and forth between locations, and put POs into a specific location etc.. with sub-locations, you can have a mix of "virtual" locations (customers) and physical locations (like shelves/rooms/...)

**Comment by eeintech** [comment]
Hello @sur5r Yes there is a misunderstanding which I will try to clarify.

[Stock ownership](https://docs.inventree.org/en/latest/stock/owner/) was implemented to prevent any other user from editing or deleting a stock item, so read-only as you mentioned.
There are two possibilities:
- stock item is assigned to single user: any other user in your InvenTree instance won't be able to edit it, except for admins and superusers.
- stock item is assigned to a group: any user outside the group won't be able to edit it, except for admins and superusers.

In your case if you assign it to a "user" representing a "customer" then you need to login as the "customer/user" owning those stock items to edit them. Or as an admin.

**Reply by sur5r** [reply]
Ok, so this is working as intended. I'm still wondering why being a member of more than one group is not possible/working. Will #2323 change this?

Edit: To clarify: My idea/intention is to use a group as a part owner for parts bought by the customer. My user is a member of multiple groups and the parts are owned by the respective group. All stock items end up as read only.

**Reply by eeintech** [reply]
Umm you're right I just reproduced it, this is a bug I believe. Do you mind creating an issue? Thanks :+1:

**Reply by sur5r** [reply]
Will do.

**Reply by sur5r** [reply]
Done #4229.
